### DISCUSSION TITLE
Add `strict` or `pedantic` mode

### DISCUSSION THREAD

**Discussion by pashagolub** [ananthakumaran/paisa#108] [Ideas]
From ledger manual:
>`--strict`
    Accounts, tags or commodities not previously declared will cause warnings.
`--pedantic`
    Accounts, tags or commodities not previously declared will cause errors.

I would like to set such mode in the configuration. Sometimes import fails or I mistype the account name. And it takes some time later to find the typo/error.

The proposal is: 
- In `strict` mode warnings appears on the Doctor page. 
- In `pedantic` mode user spots error immediately and fixes it right away.

I admit this is quite a strong limitation for most of the users, so it should be turned off by default.

If you like the idea, I may try to implement it.

**Comment by ananthakumaran** [comment]
I can make it a config. ledger has strict/pedantic, but hledger has only strict (which is the same as ledger's pedantic). Are you looking for both options or one is enough?

**Reply by pashagolub** [reply]
To have a pedantic only will be fine, I hope

**Reply by ananthakumaran** [reply]
available on master https://github.com/ananthakumaran/paisa/commit/070a1452d506a7a045af7d4647c8b3edb1391322

**Reply by pashagolub** [reply]
Thanks a lot!

**Comment by parthvzala** [comment]
does this work? I enabled strict model and all transactions started showing errors. I have defined accounts already still it raises "Unknow Account" error. I had to disable it after failed tries to fix it.

**Reply by pashagolub** [reply]
Yes, it works for me. Unfortunately error messages are not the strongest part of paisa. I suggest you to run `ledger` in console and fix all errors you will find.

**Reply by parthvzala** [reply]
Thanks. so I fixed it myself. This is what I found. 

- Paisa is not adding accounts to the ledger file.
- When we enable strict mode ledger starts throwing errors because of that
- to solve, I had to add accounts in ledger editor manually and now accounts in Paisa configuration is redundant as it doesn't matter if I have accounts at both locations or only in the ledger text. 
- Configuration accounts section only helps in the following way
  - Define account icons
  - Define credit cards with limit and automatic billing date (This feature also needs some improvements)

My ledger file is not main.ledger. I haven't checked if this is a bug or it's expected behavior.

**Reply by pashagolub** [reply]
yes, you are responsible to list accounts in ledger file. I'm not sure it's even possible to specify accounts in paisa settings

**Reply by parthvzala** [reply]
thanks. 😅 I spent some time specifying all accounts in Paisa configuration and then spent another hour figuring out why it's not working and why no issues/questions anywhere.
